### PR TITLE
chore(doc): Update docs for Gomodule support

### PR DESCRIPTION
**Pull request by vaniisgh**
**Files Changed**

- Delete Gopkg.lock & Gopkg.toml
- Change setup & developer docs
- Small change in buildscripts/build.sh

**Why is this PR required? What issue does it fix?**:
Updating docs as well as a very small fix in the build scripts to support go modules.

**What this PR does?**:
Adds the required updates.

**Any additional information for your reviewer?** :
Just a small question about this file zfs-localpv/buildscripts/zfs-driver/Dockerfile
 this line `RUN apt-get -y install rsyslog libssl-dev xfsprogs ca-certificates` breaks code on my local machine, should I expect this? and how can I fix this ? :)

**Checklist:**
- [ ] Fixes #<issue number>
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track: No
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track : No

### CODE REVIEW

**Comment by pawanpraka1** [comment]
@vaniisgh can you add change log in this PR?

**Comment by vaniisgh** [comment]
not sure why my re-base did this .. should I open a fresh PR?

**Comment by pawanpraka1** [comment]
> not sure why my re-base did this .. should I open a fresh PR?

you can fix the changes, force push(git push -f) here in the same PR only.

**Comment by vaniisgh** [comment]
it says everything is upto date there ...

File: docs/developer-setup.md [hunk 1]
```diff
@@ -15,19 +15,17 @@
...
 
-Place openebs/zfs-localpv's code on your `GOPATH` using the following cloning procedure.
+Place openebs/zfs-localpv's code on your local machine using the following cloning procedure.
 Create your clone:
 
 ```sh
-
-mkdir -p $GOPATH/src/github.com/openebs
-cd $GOPATH/src/github.com/openebs
+cd path/to/zfs-loclalpv
```
Comment (pawanpraka1) [inline docs/developer-setup.md:22]: remove this also.

File: docs/developer-setup.md [hunk 2]
```diff
@@ -15,19 +15,17 @@
...
 
 # Note: Here user= your github profile name
 git clone https://github.com/$user/zfs-localpv.git
 
 # Configure remote upstream
-cd $GOPATH/src/github.com/openebs/zfs-localpv
+cd path/to/folder/zfs-localpv
```
Comment (pawanpraka1) [inline docs/developer-setup.md:28]: make it "cd zfs-localpv"

**Comment by pawanpraka1** [comment]
@vaniisgh can you DCO sign off all the commit ? Few commits are not having DCO sign, not able to merge it.

**Comment by vaniisgh** [comment]
yes sorry about all this, my focus is not good at the moment.

**Pull request merged**